server/jetstream_test.go

```diff
@@ -14642,6 +14642,52 @@ func TestJetStreamMaxMsgsPerSubjectWithDiscardNew(t *testing.T) {
 	}
 }
 
+// Issue #2836
+func TestJetStreamInterestRetentionBug(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer s.Shutdown()
+
+	if config := s.JetStreamConfig(); config != nil {
+		defer removeDir(t, config.StoreDir)
+	}
+
+	nc, js := jsClientConnect(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:      "TEST",
+		Subjects:  []string{"foo.>"},
+		Retention: nats.InterestPolicy,
+	})
+	require_NoError(t, err)
+
+	_, err = js.AddConsumer("TEST", &nats.ConsumerConfig{Durable: "c1", AckPolicy: nats.AckExplicitPolicy})
+	require_NoError(t, err)
+
+	test := func(token string, fseq, msgs uint64) {
+		subj := fmt.Sprintf("foo.%s", token)
```

Comment:
Maybe add `t.Helper()` here so if there was a failure we would know which which call to test() is the one failing?